openmdao/utils/tests/test_assert_utils.py

```diff
@@ -82,14 +82,59 @@ def compute_partials(self, inputs, partials):
         rtol = 1.e-6
         try:
             assert_check_partials(data, atol, rtol)
-        except AssertionError as err:
-            expected_str = "error in partial of y wrt x1 in"
-            self.assertTrue(expected_str in str(err),
-                            msg="\n\nActual err msg:\n{} \n\ndoes not contain expected string:\n\n{}".format(str(err),
-                                                                                                     expected_str))
+        except ValueError as err:
+            err_string = str(err)
+            self.assertEqual(err_string.count('Assert Check Partials failed for the following Components'), 1)
+            self.assertEqual(err_string.count('1e-06'), 2)
+            self.assertEqual(err_string.count('Component:'), 1)
+            self.assertEqual(err_string.count('< output > wrt < variable >'), 1)
+            self.assertEqual(err_string.count('norm'), 2)
+            self.assertEqual(err_string.count('y wrt x1'), 4)
+            self.assertEqual(err_string.count('y wrt x2'), 4)
+            self.assertEqual(err_string.count('abs'), 6)
+            self.assertEqual(err_string.count('rel'), 6)
+            self.assertEqual(err_string.count('fwd-fd'), 4)
+            self.assertEqual(err_string.count('rev-fd'), 4)
         else:
             self.fail('Exception expected.')
 
+    def test_feature_assert_check_partials_exception_expected(self):
+        class MyComp(ExplicitComponent):
+            def setup(self):
+                self.add_input('x1', 3.0)
+                self.add_input('x2', 5.0)
+
+                self.add_output('y', 5.5)
+
+                self.declare_partials(of='*', wrt='*')
+
+            def compute(self, inputs, outputs):
+                """ Compute outputs. """
+                outputs['y'] = 3.0 * inputs['x1'] + 4.0 * inputs['x2']
+
+            def compute_partials(self, inputs, partials):
+                """Intentionally incorrect derivative."""
+                J = partials
+                J['y', 'x1'] = np.array([4.0])
+                J['y', 'x2'] = np.array([40])
+
+        prob = Problem()
+        prob.model = MyComp()
+
+        prob.set_solver_print(level=0)
+
+        prob.setup(check=False)
+        prob.run_model()
+
+        data = prob.check_partials(suppress_output=True)
+
+        atol = 1.e-6
+        rtol = 1.e-6
+        try:
+            assert_check_partials(data, atol, rtol)
+        except ValueError as err:
+            print(str(err))
```

Comment:
should this just raise the err rather than print it?

Reply:
but then the test fails and the docs don't build unless I am missing a trick to get around that

Reply:
oh sorry, I missed that this was the feature  test